lib/graphql_schema.rb

```diff
@@ -24,6 +24,10 @@ def types
     @types ||= @hash.fetch('types').map{ |type_hash| TypeDefinition.new(type_hash) }.sort_by(&:name)
   end
 
+  def types_by_name
```

Comment:
I think @dylanahsmith had some good comments here https://github.com/Shopify/graphql_schema/pull/4#discussion_r96962666
It's nice if a hash is exposed because you have all the extra functionality that hashes provide.

As for naming I'm ok calling it whatever.